data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/RecordDispatcherImpl.java

```diff
@@ -176,6 +178,17 @@ public Future<Void> dispatch(ConsumerRecord<Object, CloudEvent> record) {
             return Future.failedFuture(msg);
         }
 
+        if (record.value() instanceof OffsetSkippingCloudEvent) {
+            incrementSkippedOffset(recordContext);
+            final var msg = String.format(
+                    "Skipping offset topic %s, partition %d, offset %d",
+                    record.topic(), record.partition(), record.offset());
+            logger.debug(msg);
+            recordDispatcherListener.recordReceived(record);
+            recordDispatcherListener.recordDiscarded(record);
+            return Future.failedFuture(msg);
```

Comment:
Is there a reason why we want to fail this future? IMO, since we are just skipping it, this is more of a "success" than not...

But, I also have not looked at this code in while, so I could be wrong.

Any thoughts @maschmid @creydr ?